flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/JobMasterPartitionTrackerImplTest.java

```diff
@@ -240,54 +245,120 @@ public void testReleaseOrPromote() {
                                         createTaskExecutorGateway(
                                                 resourceId, taskExecutorReleaseCalls)));
 
+        final ResourceID taskExecutorId = ResourceID.generate();
+        final ResultPartitionID resultPartitionId1 = new ResultPartitionID();
+        final ResultPartitionID resultPartitionId2 = new ResultPartitionID();
+
+        partitionTracker.startTrackingPartition(
+                taskExecutorId,
+                AbstractPartitionTrackerTest.createResultPartitionDeploymentDescriptor(
+                        resultPartitionId1, true));
+
+        partitionTracker.startTrackingPartition(
+                taskExecutorId,
+                AbstractPartitionTrackerTest.createResultPartitionDeploymentDescriptor(
+                        resultPartitionId2, false));
+
+        assertThat(
+                partitionTracker.getAllTrackedPartitions().stream()
+                        .map(desc -> desc.getShuffleDescriptor().getResultPartitionID())
+                        .collect(Collectors.toList()),
+                hasItems(resultPartitionId1, resultPartitionId2));
```

Comment:
I think `contains` is better which requires exact matching of collections.

Reply:
`contains` is sensitive about order while `hasItems` doesn't